components/application_manager/application_manager.h

```diff
@@ -390,15 +389,16 @@ class ApplicationManager {
     uint16_t GetKeySet(const uint16_t keyCode);
 
     std::unique_ptr<SessionCallback> m_sessionCallback;
-    uint16_t m_nextAppId;
     Ait m_ait;
-    App m_app;
+    std::unordered_map<uint16_t, std::unique_ptr<App>> m_apps;
```

Comment:
Not feasible, as the copy assignment operator is deleted to avoid confusion with App ids, and is implicitly called when we  try to insert a new key-value pair to m_maps. Smart pointers seems to be the best option here.

Reply:
Why not just use the ID as the key in the map?